hoomd/filter/custom.py

```diff
@@ -0,0 +1,51 @@
+"""Contains a class for custom particle filters in Python."""
+from abc import ABCMeta, abstractmethod
+
+class CustomFilter(metaclass=ABCMeta):
+    """Abstract base class for custom particle filters.
+
+    The class allows the definition of particle filters in Python (see
+    `hoomd.filter.ParticleFilter`.
+    """
+    @abstractmethod
+    def __call__(self, state):
+        """Return the local particle tags that match the filter.
+
+        This can either return the tags that are local to an MPI rank or the
+        tags of all particles that match in the entire state.  Tag numbers in a
+        `hoomd.Snapshot` object are just their index.
```

Comment:
That and for easier working with local snapshots since accessing a particle who is outside your MPI domain would be easier with a filter that returned all matching particles.